api/types/plugin_test.go

```diff
@@ -1133,6 +1133,33 @@ func TestPluginAWSICSettings(t *testing.T) {
 				require.Equal(t, "some-oidc-integration", oidc.IntegrationName)
 			},
 		},
+		{
+			name: "empty Role Sync Mode is preserved",
+			mutateSettings: func(cfg *PluginAWSICSettings) {
+				cfg.RolesSyncMode = ""
+			},
+			assertErr: require.NoError,
+			assertValue: func(t *testing.T, cfg *PluginAWSICSettings) {
+				require.Empty(t, cfg.RolesSyncMode)
+			},
+		},
+		{
+			name: "valid Role Sync Mode is preserved",
+			mutateSettings: func(cfg *PluginAWSICSettings) {
+				cfg.RolesSyncMode = AWSICRolesSyncModeNone
+			},
+			assertErr: require.NoError,
+			assertValue: func(t *testing.T, cfg *PluginAWSICSettings) {
+				require.Equal(t, AWSICRolesSyncModeNone, cfg.RolesSyncMode, cfg.RolesSyncMode)
```

Comment:
nit: the fourth param here is the error message, passing cfg.RolesSyncMode probably isn't helpful when added to the default require.Equal error message

Reply:
Yup - just a cut & paste error